### PR TITLE
Environment variable configuration

### PR DESCRIPTION
**Pull request by tommyk**
# overview

we wanted to be able to use environment variables to drive some of the configuration instead of just args.  let me know if this works for you, or if we need to add more to this PR to get it merged.

thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,18 +15,18 @@ _Note_: the proxy should be used with care as any client that can access the API
 ## Usage
 
 [embedmd]:# (tmp/help.txt)
```
Comment (squat) [inline README.md]: The content of this code block is programmatically generated via `make README.md`, so next time that someone runs `make README.md` these changes will get blown away :/
I would suggest keeping this "simple" for now and using the generated output at the expense of showing the environment variables. We could optionally document the environment variable in the option help text.

Comment (jpweber) [reply README.md]: Is the readme change the only blocker? I will volunteer to make these changes if that will unblock this change.

**Comment by jpweber** [comment]
Reviving a Zombie. I have pushed updates to this PR to bring it inline with the current state of the project.

File: main.go [hunk 2]
```diff
@@ -70,28 +70,47 @@ type oidcConfig struct {
 	password     string
 }
 
+func LookupEnvOrString(key string, defaultVal string) string {
```
Comment (squat) [inline main.go]: Can we please make all of these lowercase, ie non-exported functions? That's the standard for Go and for the Observatorium project if they are only used internally

**Review by squat — CHANGES_REQUESTED** [review]
I have just one request otherwise this I'm happy to merge this PR

File: main.go (hunk 2 above)
Comment (jpweber) [reply main.go]: Changes have been made and pushed.

**Review by squat — APPROVED** [review]
I'm happy with the state of this for now. Tha KS for taking this over the finish line @jpweber

**Comment by jpweber** [comment]
> I'm happy with the state of this for now. Tha KS for taking this over the finish line @jpweber

Very welcome. Thanks for reviewing.

**Comment by astro-walker** [comment]
Bump this PR @jpweber @squat can we get this merged?

**Comment by squat** [comment]
wooo! @astro-walker done :))

**Comment by astro-walker** [comment]
Amazing thanks for the fast response 🙏

**Comment by tommyk** [comment]
3+ years, feels good.

**Comment by squat** [comment]
ToT we try to keep this project moving at the speed of enterprise

**Pull request merged**